### DISCUSSION TITLE
403 Permission Denied when uploading a prerelease package while one is still pending automated review

### DISCUSSION THREAD

**Discussion by hach-que** [chocolatey/home#191] [Q&A]
When my build server attempts to push a new prerelease version of a package, I get a 403 Permission Denied if there is another prerelease version currently set to "Validation Testing Pending" / "Verification Testing Pending":

```
Pushing uefs.pre.nupkg to 'https://push.chocolatey.org/'...
  PUT https://push.chocolatey.org/api/v2/package/
  Forbidden https://push.chocolatey.org/api/v2/package/ 5190ms
error: Response status code does not indicate success: 403 (Forbidden).
```

Since there's no way to delete or cancel a prerelease package before it goes through verification, there's no workaround other than waiting for the old prerelease package to (unnecessarily) go through review.

Also, this behaviour doesn't seem to be documented anywhere on the website, since I searched extensively to see if this is expected and there's no mention of it.

**Comment by hach-que** [comment]
Oh hmm. Weird. I just had it fail when the current prerelease package had passed validation and verification testing and just had "Scan Testing Pending". Though it was still unlisted at this point:

![image](https://user-images.githubusercontent.com/504826/189241528-a2002b5d-7e94-4df8-85ea-7320d977b9c6.png)

**Comment by AdmiringWorm** [comment]
As this issue is formulated more as a question/confusion, I will be moving this over to discussions instead.

**Comment by AdmiringWorm** [comment]
> When my build server attempts to push a new prerelease version of a package, I get a 403 Permission Denied if there is another prerelease version currently set to "Validation Testing Pending" / "Verification Testing Pending":

This is currently by design. As long as there is not a `stable` version that is either in an approved, or exempted state already available for the package only one version of the package can be in the moderation queue at the same time.

> Also, this behaviour doesn't seem to be documented anywhere on the website, since I searched extensively to see if this is expected and there's no mention of it.

That is a good point that we should probably document, may I ask you to open up an issue over at https://github.com/chocolatey/docs to request this being documented? (If an issue for it doesn't already exist).

> Since there's no way to delete or cancel a prerelease package before it goes through verification, there's no workaround other than waiting for the old prerelease package to (unnecessarily) go through review.

You are right, there is no way to delete or cancel a pre-release package, and can however still be unlisted once it has gone through all the moderation service checks we have in place.
This could be considered as a possible enhancement request to allow _cancelling_ a package while it is in moderation, but to do that we need an issue that has the templates we use for this repository filled out to completion first.

> Oh hmm. Weird. I just had it fail when the current prerelease package had passed validation and verification testing and just had "Scan Testing Pending". Though it was still unlisted at this point:

A package is not made listed before it has gone through all of the package services, that includes validator, verifier and the package scanner.
